drivers/gnss/gnss_u_blox_protocol/gnss_u_blox_protocol.h

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+#include <zephyr/modem/ubx.h>
+#include "gnss_u_blox_protocol_defines.h"
+
+#ifndef ZEPHYR_U_BLOX_PROTOCOL_
+#define ZEPHYR_U_BLOX_PROTOCOL_
+
+#define UBX_BAUDRATE_COUNT			9
+
+/* When a configuration frame is sent, the device requires some delay to reflect the changes. */
+/* TODO: check what is the precise waiting time for each message. */
+#define UBX_CFG_RST_WAIT_MS			6000
+#define UBX_CFG_GNSS_WAIT_MS			6000
+#define UBX_CFG_NAV5_WAIT_MS			6000
+
+extern const uint32_t ubx_baudrate[UBX_BAUDRATE_COUNT];
+
+#define UBX_FRM_GET_PAYLOAD_SZ			0
+#define UBX_CFG_ACK_PAYLOAD_SZ			2
+#define UBX_CFG_NAK_PAYLOAD_SZ			2
+#define UBX_CFG_RATE_PAYLOAD_SZ			6
+#define UBX_CFG_PRT_POLL_PAYLOAD_SZ		1
+#define UBX_CFG_PRT_POLL_FRM_SZ			UBX_FRM_SZ_WO_PAYLOAD + UBX_CFG_PRT_POLL_PAYLOAD_SZ
```

Comment:
this and the other defines with operations should be parenthesized, just to be sure there's no hidden operation priority errors